### DISCUSSION TITLE
How to mutate query params with useSWRInfinite?

### DISCUSSION THREAD

**Discussion by wassim** [vercel/swr#658] [General] [answered]
I need to update the search params;

`const { data, error, size, setSize, mutate } = useSWRInfinite(index => '/api/posts?page=${index + 1}&limit=${pageSize}&search=${params}', fetcher)`

```
const handleSearch = (params) => {
    mutate(`/api/posts`)
}
```

Does the mutate function accepts query params?

**Comment by sergiodxa** [comment, marked as answer]
The mutate function is not used to change the key, only the cached data. If you want to change the key you are using store part of it in a state and when you update your state it will be updated automatically and trigger a new fetch.

**Reply by wassim** [reply]
Thanks. I have misunderstood the mutate function. Maybe you could add an example of how to update fetch with query params in the docs? That will be handy!
